File: Star Patterns/solidDiamond.cpp
/***====================================
 *               colVals               *
 *              1 2 3 4 5              *
 *  rowVal = 1      *                  *
 *  rowVal = 2    * * *                *
 *  rowVal = 3  * * * * *              *
 *  rowVal = 4    * * *                *
 *  rowVal = 5      *                  *
 *                                     *
 ***===================================*/

#include <iostream>
using namespace std;

int main() {
    int numRows = 7;
    int rowVal;
    int colVal;
    int numSpaces = numRows / 2;
    int numPatternSymbols = 1;

    char patternSymbol = '*';

    for(rowVal = 1; rowVal <= numRows; rowVal++) {
        for(colVal = 1 ;colVal <= numSpaces; colVal++) {
            cout<<"  ";
        } 
        for(colVal = 1; colVal <= numPatternSymbols; colVal++) {
            cout<<patternSymbol<<" ";
        }
        if(rowVal <= numRows / 2) {
            numSpaces--;
            numPatternSymbols += 2;
        } else {
            numSpaces++;
            numPatternSymbols -= 2;
        }
        cout<<"\n";
    }
    return 0;
}

File: Star Patterns/solidDiamond.py
numRows = 7

# floor division to get only integer value.
numSpaces = numRows // 2;   
numPatternSymbols = 1;

patternSymbol = '*'

for rowVal in range(1, numRows + 1):
    for colVal in range(1, numSpaces + 1):
        print("  ", end="")
    for colVal in range(1, numPatternSymbols + 1):
        # heer, I am only using this "" because, 
        # sep parameter is already giving us one space as sep=" "
        print(patternSymbol, "", end="")
    if rowVal <= numRows // 2:
        numPatternSymbols += 2
        numSpaces -= 1
    else:
        numPatternSymbols -= 2
        numSpaces += 1
    print()   # for new line
